tests/scalers/couchdb/couchdb_test.go

```diff
@@ -0,0 +1,394 @@
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:alpine
+`
+
+	secretTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  connectionString: {{.Base64Connection}}
+  username: {{.Username}}
+  password: {{.Password}}
+`
+	triggerAuthTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: password
+    name: {{.SecretName}}
+    key: password
+  - parameter: username
+    name: {{.SecretName}}
+    key: username
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: 1
+  maxReplicaCount: 2
+  triggers:
+  - type: couchdb
+    metadata:
+      host: {{.HostName}}
+      port: "5984"
+      dbName: "animals"
+      queryValue: "1"
+      query: '{ "selector": { "feet": { "$gt": 0 } }, "fields": ["_id", "feet", "greeting"] }'
+      activationQueryValue: "1"
+      metricName: "global-metric"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+`
+)
+
+func CreateKubernetesResourcesCouchDB(t *testing.T, kc *kubernetes.Clientset, nsName string, data interface{}, templates []Template) {
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+}
+
+func TestCouchDBScaler(t *testing.T) {
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	// setup couchdb
+	CreateNamespace(t, kc, testNamespace)
+	installCouchDB(t)
+	
+	data, templates := getTemplateData(kc)
+	CreateKubernetesResourcesCouchDB(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	// test scaling
+	testActivation(t, kc)
+	testScaleUp(t, kc)
+	testScaleDown(t, kc)
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func generateUUID() string {
+	id := strings.ReplaceAll(uuid.New().String(), "-", "")
+	return id
+}
+
+func installCouchDB(t *testing.T) {
+	_, err := ExecuteCommand(fmt.Sprintf("helm repo add couchdb %s", couchdbHelmRepo))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	_, err = ExecuteCommand("sudo helm repo update")
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+	uuid := generateUUID()
+	_, err = ExecuteCommand(fmt.Sprintf("sudo helm install test-release  --set couchdbConfig.couchdb.uuid=%s --namespace %s couchdb/couchdb", uuid, testNamespace))
```

Comment:
Do we need sudo here? [e2e test fails](https://github.com/kedacore/keda/actions/runs/3456732803/jobs/5769756643#step:8:273) because sudo isn't installed in the agents